state/connection/ClientState.kt

```diff
@@ -91,21 +92,21 @@ class ClientState constructor(
     ) = tryConnectAsync(address, null, onSuccess) { TypeDB.coreClient(address) }
 
     fun tryConnectToTypeDBClusterAsync(
-        address: String, username: String, password: String,
+        addresses: Set<String>, username: String, password: String,
         tlsEnabled: Boolean, onSuccess: () -> Unit
-    ) = tryConnectToTypeDBClusterAsync(address, username, TypeDBCredential(username, password, tlsEnabled), onSuccess)
+    ) = tryConnectToTypeDBClusterAsync(addresses, username, TypeDBCredential(username, password, tlsEnabled), onSuccess)
 
     fun tryConnectToTypeDBClusterAsync(
-        address: String, username: String, password: String,
+        addresses: Set<String>, username: String, password: String,
         caPath: String, onSuccess: () -> Unit
     ) = tryConnectToTypeDBClusterAsync(
-        address, username, TypeDBCredential(username, password, Path.of(caPath)), onSuccess
+        addresses, username, TypeDBCredential(username, password, Path.of(caPath)), onSuccess
     )
 
     private fun tryConnectToTypeDBClusterAsync(
-        address: String, username: String,
+        addresses: Set<String>, username: String,
         credentials: TypeDBCredential, onSuccess: () -> Unit
-    ) = tryConnectAsync(address, username, onSuccess) { TypeDB.clusterClient(address, credentials) }
+    ) = tryConnectAsync(addresses.first(), username, onSuccess) { TypeDB.clusterClient(addresses, credentials) }
```

Comment:
The reason `TypeDB.clusterClient` supports inputting multiple addresses is so that in case any one of the nodes is down, it can attempt to connect to the other addresses instead until it finds one that is alive.

The logic as currently written here is counter-intuitive; it _looks_ at a glance like it's only attempting to connect the first address. This is however untrue. It behaves fine, but it looks odd. What is `address`, anyway? Surely we should be storing a list of addresses in `ClientState`, not a single one? The single one doesn't make sense for Cluster.

Reply:
This is a little weird. This argument in `tryConnectAsync` is named `newAddress`, which only really sets `address` under `//state/connection/ClientState`. The only result of this is to make an address available to the UI in the top right corner once we've connected to TypeDB.

Without some sort of identity string for clusters, we'll ultimately only end up displaying one or two IPs in the corner, which doesn't really inform the user of much. Does `serverIdentityString` make more sense as a name here?

FYI: the real addresses for a cluster client are stored under the hood in an inaccessible property of `ClusterClient` in `typedb-client-java`.

Reply:
Indeed. It's not particularly informative right now. As discussed in person, it's quite a significant chunk of work creating named connection configurations, which would be by far the most informative name for a connection. So sticking with `addresses.first()` is fine.

However, we should certainly rename (and name) the parameter for clarity. How about `connectionName`?

Reply:
Done.